serving/using-cert-manager-on-gcp.md

```diff
@@ -0,0 +1,228 @@
+  --from-file=key.json=$HOME/key.json
+
+# Delete the local secret
+rm ~/key.json
+
+```
+
+## Configure CertManager to use your DNS admin service account
+
+Next, we'll configure cert-manager to request new certificates and 
+verify the challenges using DNS.
+
+### Specify a certificate issuer
+
+For this example, we're going to configure cert-manager to use 
+LetsEncrypt, however you can use any certificate provider that
+supports the ACME protocol.
+
+Here we configure to use the `dns01` challenge type, which will
+enable certificate generation and wildcard certificates.
+
+```shell
+kubectl apply -f - <<EOF
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: ClusterIssuer
+metadata:
+  name: letsencrypt-issuer
+  namespace: cert-manager
+spec:
+  acme:
+    server: https://acme-v02.api.letsencrypt.org/directory
+    email: myemail@gmail.com
+    privateKeySecretRef:
+      # Set privateKeySecretRef to any unused secret name.
+      name: letsencrypt-issuer
+    dns01:
+      providers:
+      - name: cloud-dns-provider
+        clouddns:
+          # Set this to your GCP project-id
+          project: $PROJECT_ID
+          # Set this to the secret that we publish our service account key
+          # in the previous step.
+          serviceAccountSecretRef:
+            name: cloud-dns-key
+            key: key.json
+EOF
+
+```
+
+To check if your ClusterIssuer is valid, run
+
+```shell
+kubectl get clusterissuer -n cert-manager letsencrypt-issuer -o yaml
+```
+
+and confirm that its conditions have `Ready=True`.  For an example:
+
+```yaml
+status:
+  acme:
+    uri: https://acme-v02.api.letsencrypt.org/acme/acct/40759665
+  conditions:
+  - lastTransitionTime: 2018-08-23T01:44:54Z
+    message: The ACME account was registered with the ACME server
+    reason: ACMEAccountRegistered
+    status: "True"
+    type: Ready
+```
+
+### Specifying our certificate
+
+Next we need to configure which certificate issuer to use
+and which secret we'll publish the certificate into.
+
+```shell
+# Change this value to the domain you want to use.
+export DOMAIN=your-domain.com
+
+kubectl apply -f - <<EOF
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Certificate
+metadata:
+  name: my-certificate
+  # Istio certs secret lives in the istio-system namespace, and
+  # a cert-manager Certificate is namespace-scoped.
+  namespace: istio-system
+spec:
+  # Reference to the Istio default cert secret.
+  secretName: istio-ingressgateway-certs
+  acme:
+    config:
+    # Each certificate could comprise of different ACME challenge
+    # solver.  In this example we are using one provider for all
+    # the domains.
+    - dns01:
+        provider: cloud-dns-provider
+      domains:
+      # Since certificate wildcards only allow one level, we will
+      # need to one for every namespace.  We don't need to use
+      # wildcard here, fully-qualified domains will work fine too.
+      - "*.default.$DOMAIN"
+      - "*.other-namespace.$DOMAIN"
```

Comment:
This will need to be updated for each namespace that knative is used in, correct?

Reply:
Done